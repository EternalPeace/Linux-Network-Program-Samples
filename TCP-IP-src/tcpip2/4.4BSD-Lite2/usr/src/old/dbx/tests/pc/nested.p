(*
 * Test of nested procedures and modules.
 *)

program nested (input, output);
var
    i, k : integer;

procedure p (var i : integer);
var
    j, k : integer;

    procedure nestedp (var j : integer);
    var
	i : integer;
    begin
	i := j + 2;
	j := i;
    end;

begin
    j := i + 1;
    nestedp(j);
    i := j;
end;

begin
    i := 3;
    p(i);
end.
