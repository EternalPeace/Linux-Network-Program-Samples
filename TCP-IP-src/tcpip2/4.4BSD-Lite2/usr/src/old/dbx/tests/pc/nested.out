reading symbolic information ...
p.i p.nestedp.i nested.i
p.j p.nestedp.j
[1] stop at 18
[1] stopped in nestedp at line 18
   18   	j := i;
p.k
6 4 3 3 4 
stopped in nestedp at line 19
   19       end;
stopped in p at line 24
   24       i := j;
3 6 3 

execution completed
6 

